packages/sanity/src/structure/comments/src/hooks/useResolveCommentsEnabled.ts

```diff
@@ -1,24 +1,38 @@
 import {useMemo} from 'react'
 import {getPublishedId, useFeatureEnabled, useSource} from 'sanity'
 
+interface ResolveCommentsEnabled {
+  enabled: boolean
+  reason: 'upsell' | null
+}
+
 /**
  * @internal
  * A hook that resolves if comments are enabled for the current document and document type
  * and if the feature is enabled for the current project.
  */
-export function useResolveCommentsEnabled(documentId: string, documentType: string): boolean {
+export function useResolveCommentsEnabled(
+  documentId: string,
+  documentType: string,
+): ResolveCommentsEnabled {
   // Check if the projects plan has the feature enabled
   const {enabled: featureEnabled, isLoading} = useFeatureEnabled('studioComments')
 
   const {enabled} = useSource().document.unstable_comments
-
   // Check if the feature is enabled for the current document in the config
   const enabledFromConfig = useMemo(
     () => enabled({documentType, documentId: getPublishedId(documentId)}),
     [documentId, documentType, enabled],
   )
+  const value = useMemo(
+    () => ({
+      enabled: !isLoading && enabledFromConfig,
```

Comment:
This value needs to include the `featureEnabled` value as well. Currently, it only checks the enabled value resolved from the config API.

Reply:
This is intentional to support disabling comments from config.

You could have 3 type of values here:
```
 {enabled:false, reason: null}  //  Disabled from config, won't show anything related to comments.- Will remove reason from here to make it simpler to follow.
 {enabled: true, reason: null}  // Current UI, no changes
 {enabled: true, reason:  "upsell} // Shows upsell UI.
```

Pushed an update here to clarify.